src/target.cc

```diff
@@ -0,0 +1,84 @@
+#include "target.h"
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "logging/logging.h"
+
+bool Target::hasTag(const Uptane::Target& target, const std::vector<std::string>& tags) {
+  if (tags.empty()) {
+    return true;
+  }
+
+  const auto target_tags = target.custom_data()[TagField];
+  for (Json::ValueConstIterator ii = target_tags.begin(); ii != target_tags.end(); ++ii) {
+    auto target_tag = (*ii).asString();
+    if (std::find(tags.begin(), tags.end(), target_tag) != tags.end()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void Target::setCorrelationID(Uptane::Target& target) {
+  std::string id = target.custom_version();
+  if (id.empty()) {
+    id = target.filename();
+  }
+  boost::uuids::uuid tmp = boost::uuids::random_generator()();
+  target.setCorrelationId(id + "-" + boost::uuids::to_string(tmp));
+}
+
+std::string Target::ostreeURI(const Uptane::Target& target) {
+  std::string uri;
+  auto uri_json = target.custom_data().get("compose-apps-uri", Json::Value(Json::nullValue));
+  if (!uri_json.empty()) {
+    uri = uri_json.asString();
+  }
+  return uri;
+}
+
+Json::Value Target::appsJson(const Uptane::Target& target) {
+  return target.custom_data().get(Target::ComposeAppField, Json::Value(Json::nullValue));
+}
+
+std::string Target::appsStr(const Uptane::Target& target, boost::optional<std::vector<std::string>> app_shortlist) {
+  std::vector<std::string> apps;
+  for (const auto& app : Target::Apps(target)) {
+    if (!app_shortlist ||
+        (*app_shortlist).end() != std::find((*app_shortlist).begin(), (*app_shortlist).end(), app.name)) {
+      apps.push_back(app.name);
+    }
+  }
+  return boost::algorithm::join(apps, ",");
+}
+
+void Target::setAppsJson(Uptane::Target& target, const Json::Value& apps_json) {
+  Json::Value custom_data = target.custom_data();
+  if (!custom_data.isNull()) {
```

Comment:
seems odd. Should this method fail if its not able to set this property or throw an exception? It seems an exceptional situation to have a target with no custom data defined?

Also - this only seems to be called from `createTarget`. I don't see that referenced anywhere. Is this dead code for now we don't yet need?

Reply:
It's needed in the future :). The final "simplified" version makes use of it. I can remove it from here and put it back again, although it's kind of a waste of time...

Reply:
that's fine. how about the null check thought?

Reply:
yep, that's rather an exceptional situation, will add throwing an exception here.

Reply:
I actually removed the currently unused code since it's not possible to verify it. Will added it once it becomes needed.